### DISCUSSION TITLE
Use Regex in --proxy-domain

### DISCUSSION THREAD

**Discussion by sevetseh28** [coder/code-server#6933] [Q&A]
Hi! Im having an issue with the `{{port}}-{{host}}` value in proxy domain arg.

My host looks like:   `[a-z0-9]{8}-hello.example.net`

So an example is `1a2b3c4d-hello.example.net`.

However,I have issues when the first component is all digits because it interprets it as the port:
`12345678-hello.example.net` and I'm using the `{{port}}-{{host}}` in proxy domain.

I looked at the source code here https://github.com/coder/code-server/blob/main/src/node/routes/domainProxy.ts
but I'm not entirely sure if regex are supported there. I see that the issue is that it will transform it into `(\d+)-(.+)` regex, which is wrong for my case (I need something more strict).

I tried `{{port}}-[a-z0-9]{8}-hello.example.net` but then when accessing the IDE I got a message :

```
The workbench failed to connect to the server (Error: "//{{port}}-[a-f0-9]{8})-hello.example.net" cannot be parsed as a URL.)
```

Thank you!

**Comment by code-asher** [comment]
Oh yeah hmm this is an interesting edge case.  We escape the value so it is not possible to use regex in `--proxy-domain`.  Are you able to use a different delimiter?  Like `{{port}}+{{host}}` maybe?

If that does not work, maybe we will need to add an option to specify a regex.  `--proxy-domain-regex` or something like that.
